plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -398,27 +411,30 @@ private TestExecutionResult testPluginAgainst(MavenCoordinates coreCoordinates,
                     System.out.println("Deleting working directory "+pluginCheckoutDir.getAbsolutePath());
                     FileUtils.deleteDirectory(pluginCheckoutDir);
                 }
+
                 pluginCheckoutDir.mkdir();
                 System.out.println("Created plugin checkout dir : "+pluginCheckoutDir.getAbsolutePath());
 
                 if (localCheckoutProvided()) {
                     if (!onlyOnePluginIncluded()) {
-                        throw new RuntimeException("You specified a local clone but did not choose only one plugin to execute PCT against it");
+                        File localCheckoutPluginDir = new File(config.getLocalCheckoutDir(), plugin.name);
+                        File pomLocalCheckoutPluginDir = new File(localCheckoutPluginDir, "pom.xml");
+                        if(pomLocalCheckoutPluginDir.exists()) {
+                            System.out.println("Copy plugin directory from : " + localCheckoutPluginDir.getAbsolutePath());
+                            FileUtils.copyDirectoryStructure(localCheckoutPluginDir, pluginCheckoutDir);
+                        } else {
+                            cloneFromSCM(pomData.getConnectionUrl(), plugin.name, plugin.version, pluginCheckoutDir);
+                        }
+                    } else {
+                        // TODO this fails when it encounters symlinks (e.g. work/jobs/…/builds/lastUnstableBuild),
+                        // and even up-to-date versions of org.apache.commons.io.FileUtils seem to not handle links,
+                        // so may need to use something like http://docs.oracle.com/javase/tutorial/displayCode.html?code=http://docs.oracle.com/javase/tutorial/essential/io/examples/Copy.java
+                        System.out.println("Copy plugin directory from : " + config.getLocalCheckoutDir().getAbsolutePath());
+                        FileUtils.copyDirectoryStructure(config.getLocalCheckoutDir(), pluginCheckoutDir);
```

Comment:
Maybe it's worth do the same validation as https://github.com/jenkinsci/plugin-compat-tester/pull/62/files#diff-75e1583cf2872b928bd6bfe8c3b8f595R422 and in case the `pom.xml` file doesn't exist then run `cloneFromSCM` ? Just in case, those plugin folders are corrupted somehow

Reply:
I think that would not be a valid use case: PCT provided a way for you to use your local clone. The state of such local clone should be the caller responsibility.

When you are including several plugins it does make sense because you could not be willing to use local clones for each one.

Reply:
I've tried to be non-intrusive. At present time, if the plugin folder is corrupted the execution is marked with an error and I have keep this behaviour.